### PR TITLE
galileo: Isolated Memory Region support

### PR DESCRIPTION
**Pull request by mdlemay**
Depends on #1444.

This patchset adds drivers for the Intel Quark X1000 SoC message bus and Isolated Memory Regions (IMRs).  It also adds support for optionally configuring and locking the IMRs during boot to block DMA to memory regions that do not contain any data that needs to be DMA-accessible.

### CODE REVIEW

**Comment by mdlemay** [comment]
@aguedes, @jeez: Can you please review this PR?

**Comment by jeez** [comment]
@mdlemay I'm having a first pass on this right now. At this first moment I will only review how the feature-set is sitting in our platform, ok?

On a second pass I will review the feature per se.

**Comment by jeez** [comment]
I'm happy with the PR in general, thanks! Just a few minor points to address.
I still want to have a deeper look at the feature per se, will try my best and do it this week.

After this PR, however, I do think we need to start considering a centralized mechanism like [platform,soc]-conf.h . It is getting hard to follow how to disable a certain platform feature/peripheral, so we need to put this in place before it is too late.

**Comment by mdlemay** [comment]
Thanks for the feedback, @jeez!  I pushed a new revision of the patches just now.

I look forward to hearing more about the configuration header file idea.  I can already imagine certain challenges, such as how to change settings in Makefiles based on definitions in the headers, but I'd like to hear the basic idea first before jumping into those details.

**Comment by jeez** [comment]
LGTM !

@mdlemay re the configuration header idea, I will try to open an issue later so we can discuss it, ok?

thanks!

**Comment by nvt** [comment]
Now that #1444 is merged, what remains to be done on this PR? Currently, there are some conflicts.

**Comment by mdlemay** [comment]
I just rebased it, so I think all that remains now is for @aguedes to review it.

**Comment by mdlemay** [comment]
Actually, @aguedes told me that we should not wait for him to review these patches.  So, please feel free to merge them now.  Thanks!

**Comment by aguedes** [comment]
Just for the record, I just reviewed this PR and LGTM :+1: 

Thanks, Michael!

**Comment by nvt** [comment]
:+1:
